protocols/identify/src/lib.rs

```diff
@@ -69,6 +71,10 @@ impl MisbehaveResult {
 
 /// The trait to communicate with underlying peer storage
 pub trait Callback: Clone + Send {
+    /// Received custom message
+    fn received_identify(&mut self, context: &mut ProtocolContextMutRef, identify: Vec<u8>);
```

Comment:
it's more common to pass slice rather than vec when you just want to provide a read access
```suggestion
    fn received_identify(&mut self, context: &mut ProtocolContextMutRef, identify: &[u8]) {
```